File: 200_java/HelloWorld.java
import java.io.FileWriter;
import java.io.IOException;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class HelloWorld {
    public static void main(String[] args) {
        // Get the current date and time
        LocalDateTime currentDateTime = LocalDateTime.now();
        // Format the date and time
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
        String formattedDateTime = currentDateTime.format(formatter);

        // Define the message to log
        String message = "Hello, Java World! Current Date and Time: " + formattedDateTime;

        // Print the message to the console
        System.out.println(message);

        // Write the message to a log file
        try (FileWriter logFile = new FileWriter("hello_world.log", true)) {
            logFile.write(message + "\n");
            System.out.println("Log written to hello_world.log");
        } catch (IOException e) {
            System.out.println("An error occurred while writing to the log file.");
            e.printStackTrace();
        }
    }
}

// compile - javac HelloWorld.java
// run to create class file - java HelloWorld
// include manifest.txt with this text "Main-Class: HelloWorld" to point to entry class
// create jar file from class file - jar cfm HelloWorld.jar manifest.txt HelloWorld.class
// run the jar file - java -jar HelloWorld.jar

File: 400_r/hello_world.R
# Get the current date and time
current_datetime <- Sys.time()

# Format the date and time
formatted_datetime <- format(current_datetime, "%Y-%m-%d %H:%M:%S")

# Print the formatted date and time
cat("Hello, R World! Current date and time:", formatted_datetime, "\n")

# Specify the log file name
log_file <- "hello_world.txt"

# Write the formatted date and time to the log file
write(formatted_datetime, file = log_file, append = TRUE)

# Add a newline after each entry
write("\n", file = log_file, append = TRUE)

# Terminal - Rscript hello_world.R
# Emacs REPL - Install Emacs Speaks Statistics (EWW) package
# M-x R (run REPL)
# source("hello_world.R")
